tap_mambu/tap_generators/audit_trail_generator.py

```diff
@@ -1,58 +1,85 @@
         if type(audit_trail_bookmark) is list:
             if len(audit_trail_bookmark) != 2:
                 raise ValueError("Cannot parse audit trail bookmark from list because we expect 2 values!")
-            self.audit_trail_offset = audit_trail_bookmark[1]
             audit_trail_bookmark = audit_trail_bookmark[0]
 
-        self.endpoint_params = {
-            "sort_order": "asc",
-            "occurred_at[gte]": datetime_to_utc_str(str_to_datetime(audit_trail_bookmark)),
-            "occurred_at[lte]": datetime_to_utc_str(utc_now()),
-        }
+        self.start_datetime = str_to_localized_datetime(audit_trail_bookmark)
+        self.end_datetime = str_to_localized_datetime(datetime_to_utc_str(utc_now() + timedelta(days=1)))
 
     def _init_params(self):
         super(AuditTrailGenerator, self)._init_params()
-        # Initializing offset with value from bookmark, so we skip N records that were extracted after the bookmark
-        self.offset = self.audit_trail_offset
+        # Initializing has_more and next_start_time
+        self.has_more = True
+        self.next_start_time = None
+
+    def get_default_start_value(self):
+        audit_trail_bookmark = get_bookmark(self.state, self.stream_name, self.sub_type, self.start_date)
+
+        # Backwards compatibility check, such that it doesn't fail when we use a bookmark from an older Tap version
+        if type(audit_trail_bookmark) is list:
+            if len(audit_trail_bookmark) != 2:
+                raise ValueError("Cannot parse audit trail bookmark from list because we expect 2 values!")
+            audit_trail_bookmark = audit_trail_bookmark[0]
+        
+        return audit_trail_bookmark
+
+    def _all_fetch_batch_steps(self):
+        # Large buffer size can impact memory utilization of connector
+        # so empty the buffer once it reaches default max limit
+        if len(self.buffer) > self.max_buffer_size:
+            return
+
+        self.prepare_batch()
+        LOGGER.info(f'(generator) Stream {self.stream_name} - Fetching batch with params: {self.endpoint_params}')
+        response = self.fetch_batch()
+        # Extracting "events" field from batch, as we are returned a dict instead of a list of records
+        events_batch = response["events"]
+        self.buffer = transform_json(events_batch, self.stream_name)
+        if not self.buffer:
+            LOGGER.warning(f'(generator) Stream {self.stream_name} - NO TRANSFORMED DATA RESULTS')
+
+        self.has_more = response.get("hasMore")
+        self.next_start_time = response.get("nextStartTime")
+        LOGGER.info(f'(generator) Stream {self.stream_name} - hasMore: {self.has_more}, nextStartTime: {self.next_start_time}')
 
     def next(self):
         if not self.buffer:
-            if self.last_batch_size < self.limit:
+            if self.has_more or self.start_datetime < self.end_datetime:
                 raise StopIteration()
```

Comment:
This condition looks to me inverted. Shouldn't it raise when there's NO more data?